src/kernel/boxes/mem-collect.lisp

```diff
@@ -96,48 +96,37 @@
     'box-output :box self :value NIL
     :name "previous value(s)")))
 
-
-;;; ALWAYS IN "EV-ONCE" MODE
-
 (defmethod omNG-box-value ((self OMMemoryBox) &optional (numout 0))
-  
-  (unless (equal (ev-once-flag self) (get-ev-once-flag *ev-once-context*))
-    
-    (let ((inval (omng-box-value (car (inputs self))))
-          (size (omng-box-value (cadr (inputs self)))))
-      
+  (when (= numout 0) ;only update values if the first output is evaluated.
```

Comment:
I think this is the only relevant lines for what this commit says it does.
Below, it is not clear whether you want to cleanup or change behaviour (which I think you, see other comment).